docs/API.md

```diff
@@ -274,6 +274,10 @@ Emitted after the player enters the PLAY protocol state and can send and recieve
 
 Called when an error occurs within the client. Takes an Error as parameter.
 
+### `noCredentials` event
+
+Emitted when client didn't supply credentials and server is in online mode (authenticating is required).
```

Comment:
Im not sure if *appears* needs to be added here. Is it possible to encounter a case where `noCredentials` would be emitted and the connection is successful?

Reply:
Do you have a link to nmp/vanilla code or somewhere to doc how this gets triggered? It don't understand how packet.serverId relates to auth.

Reply:
>  I don't understand how packet.serverId relates to auth.

i don't know why there is packet.serverId check, sorry. That's why I raised the question below. Or did you mean something else? I wish we could improve clarity here. This is what I want to get: https://github.com/PrismarineJS/node-minecraft-protocol/pull/1283/files#r1454466037

Reply:
Yes, I mean we should not be formally emitting something like "no credentials" without being certain there is some credential issue and what's causing this. Usually this is what unit tests are supposed to catch. I don't think we need tests for this change, but an understanding of from where this issue originates, not just some debug comment.

Reply:
> Yes, I mean we should not be formally emitting something like "no credentials" without being certain there

Ok, I see your point. I have only my personal experience: whenever I try to connect to any online server I get it.
Not sure what to do here for now..